examples/rvc-app/rvc-common/include/rvc-service-area-delegate.h

```diff
@@ -0,0 +1,125 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/service-area-server/service-area-server.h>
+#include <app/util/config.h>
+#include <cstring>
+#include <utility>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+class RvcDevice;
+
+namespace ServiceArea {
+
+class RvcServiceAreaDelegate : public Delegate
+{
+private:
+    // containers for array attributes.
+    std::vector<ServiceArea::LocationStructureWrapper> mSupportedLocations;
+    std::vector<ServiceArea::MapStructureWrapper> mSupportedMaps;
+    std::vector<uint32_t> mSelectedLocations;
```

Comment:
is this supposed to be a std::set or does that cause additional problems?

Reply:
I see this may cause problems given that our nterfaces use "indices". Marking resolved.